### PR TITLE
Ani/enhance/t4 firebase auth poc

### PR DESCRIPTION
**Pull request by Animansoubi**
I've investigated Firebase Authentication, and we can easily have auth with email and password.
@moyarsuccess @idaosk

### CODE REVIEW

File: app/src/main/java/com/gathering/android/SignInActivity.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+        val currentUser = auth.currentUser
+        if (currentUser != null) {
+            val intent = Intent(this, MainActivity::class.java)
+            startActivity(intent)
+        } else {
+            auth.signInWithEmailAndPassword("animansoubi@gmail.com", "@nahidM107894")
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/SignInActivity.kt:30]: This will be hide under model layer interfaces and classes later.

**Review by moyarsuccess — COMMENTED** [review]
Nice, I liked this PR, now we know how we can use firebase in action. 💪

File: app/build.gradle [hunk 2]
```diff
@@ -34,11 +35,21 @@ android {
...
+
+    //Test
     testImplementation 'junit:junit:4.13.2'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.5'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
+
+    // Firebase BoM
+    implementation platform('com.google.firebase:firebase-bom:31.2.2')
```
[the remark this answers is not in the document]
Comment (Animansoubi) [reply app/build.gradle:50]: Yup, All of them are essential :) 
thanks for your feedback. @moyarsuccess

File: app/src/main/java/com/gathering/android/SignInActivity.kt (hunk 1 above)
Comment (Animansoubi) [reply app/src/main/java/com/gathering/android/SignInActivity.kt:30]: I just mocked the username and password for just checking the functionality

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -3,16 +3,15 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".MainActivity">
```
Comment (idaosk) [inline app/src/main/res/layout/activity_main.xml:7]: I don't understand why there is MainActivity class, and it's XML file. what is MainActivity supposed to do and how is it relevant to IntroActivity?

**Review by idaosk — APPROVED** [review]

**Pull request closed without merging**